### PR TITLE
Detect empty links and report accordingly

### PR DESCRIPTION
**Pull request by MiguelRocha**
Reports a more appropriate message when zimcheck finds empty links inside an article.

### CODE REVIEW

**Comment by MiguelRocha** [comment]
@kelson42 output:

```
...
 full_by_title.js (idx 70) and auth_41984_lang_ar_by_title.js (idx 3)
 lang_ar_by_title.js (idx 85) and auth_41984_lang_ar_by_title.js (idx 3)
[ERROR] Invalid internal links found:
  Found 2 of empty links in article: A/Home.html
[INFO] Overall Test Status: Fail

```

**Comment by kelson42** [comment]
@MiguelRocha Thank you, output looks good to me.

**Comment by kelson42** [comment]
@MiguelRocha A unit test would be welcome here I believe

**Comment by mgautierfr** [comment]
> A unit test would be welcome here I believe

Yes, but it is complicated to test is right now as there is not free function to test.

It would be nice to move the code checking for the article content in one (or several) function(s) working only on the content and not have one function mixing looping hover the articles and checking them.

But I'm not sure we should have this change in this PR.

**Comment by MiguelRocha** [comment]
@mgautierfr agree. Also after https://github.com/openzim/zim-tools/pull/144 is merged we can startup pushing the unit tests very easly :)

File: src/zimcheck.cpp [hunk 1]
```diff
@@ -437,23 +437,38 @@ void test_articles(const zim::File& f, ErrorLogger& reporter, ProgressBar progre
...
+                    nremptylinks++;
+                    continue;
+                }
 
                 if (isOutofBounds(l, baseUrl))
                 {
                     std::ostringstream ss;
                     ss << l << " is out of bounds. Article: " << it->getLongUrl();
                     reporter.addError(TestType::URL_INTERNAL, ss.str());
+                    reporter.setTestResult(TestType::URL_INTERNAL, false);
```
Comment (mgautierfr) [inline src/zimcheck.cpp:456]: Is this really related to empty links detection ?

If not we should move it to a different commit.

**Review by mgautierfr — CHANGES_REQUESTED** [review]

File: src/zimcheck.cpp (hunk 1 above)
Comment (MiguelRocha) [reply src/zimcheck.cpp:314]: Done

**Review by mgautierfr — APPROVED** [review]

**Comment by MiguelRocha** [comment]
@kelson42 I think we can merge this one. I'm also working on the full automated code for zimcheck in this PR: https://github.com/openzim/zim-tools/pull/153/files
